kong/runloop/certificate.lua

```diff
@@ -119,15 +119,23 @@ local function fetch_sni(sni, i)
 end
 
 
-local function fetch_certificate(sni)
-  local certificate, err = singletons.db.certificates:select(sni.certificate)
+local function fetch_certificate(pkey, sni_name)
+  local certificate, err = singletons.db.certificates:select(pkey)
   if err then
-    return nil, "failed to fetch certificate for '" .. sni.name .. "' SNI: " ..
-                err
+    if sni_name then
+      return nil, "failed to fetch certificate for '" .. sni_name .. "' SNI: " ..
+                  err
+    end
+
+    return "failed to fetch certificate " .. pkey.id
```

Comment:
A big mistake here I believe: missing `return nil` (first return value)? This would make the above string the valid, cached return value of `fetch_certificate`

Reply:
Good catch @thibaultcha, that was fixed!